release-python-poetry/action.yaml

```diff
@@ -0,0 +1,140 @@
+name: Reusable workflow for building and publishing python packages to TestPyPI or PyPI
+
+on:
+  workflow_call:
+    inputs:
+      scope:
+        description: "Scope of the release; See: https://python-poetry.org/docs/cli/#version"
+        required: true
+        type: string
+      python-version:
+        description: "The python version for setting up poetry."
+        required: false
+        default: 3.7
+        type: number
+      publish-to-test:
+        description: "If set to true, the packages are published to test.pypi.org"
+        required: false
+        default: false
+        type: boolean
+      ref:
+        description: "The ref name to checkout the repository."
+        required: true
+        type: string
+    secrets:
+      github-username:
+        description: "The github username for pushing."
+        required: true
+      github-email:
+        description: "The github email for pushing."
+        required: true
+      github-token:
+        description: "The github token for pushing."
+        required: true
+      pypi-token:
+        description: "The pypi token for pushing packages."
+        required: true
+
+    # Map the workflow outputs to job outputs
+    outputs:
+      release-tag:
+        description: "The release tag."
+        value: ${{ jobs.create-release.outputs.release-tag }}
+      old-tag:
+        description: "The old tag."
+        value: ${{ jobs.create-release.outputs.old-tag }}
+
+jobs:
+  create-release:
+    runs-on: ubuntu-20.04
+    env:
+      PYTHON_VERSION: ${{ inputs.python-version }}
+    # Map the job outputs to step outputs
+    outputs:
+      release-tag: ${{ steps.release-tag.outputs.tag }}
+      old-tag: ${{ steps.release-tag.outputs.oldTag }}
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+        with:
```

Comment:
I think it would be good to allow other options like `fetch-depth` through an input parameter here as well